### PR TITLE
KNN Classifier

### PR DESCRIPTION
**Pull request by ankurshukla03**
- added only one KNN notebook for lm3 file
- Have a look at solution-lm3-KNN notebook. If this is the format we are expecting then i will do the same for each classifier

### CODE REVIEW

**Comment by tristaaan** [comment]
Great job! ~~Deleting the omnibus lm3-classifier file might have been jumping the gun a bit (would do that last so you can keep referencing it)~~ Looks like creating new classifier files from here out will be much easier!

**Comment by tristaaan** [comment]
Edited last comment, didn't see that it _was_ added back. LGTM

**Review by tristaaan — APPROVED** [review]

**Comment by diegocasmo** [comment]
@ankurshukla03 @tristaaan 

Is ``get_train_test_split`` repeated in all notebooks? Could be useful to place it in just one place, instead of re-declaring it in each notebook

**Comment by diegocasmo** [comment]
@ankurshukla03 @tristaaan 

As a matter of fact, it might be even cooler if instead of returning ``(train_df, test_df)`` from ``get_train_test_split``, return instead ``(train_data, train_labels, test_data, test_labels)``. This way we don't have to unpack those values every time (this logic also seems to be repeated across all notebooks)

**Comment by ankurshukla03** [comment]
yeah i decided to add that file back if in by any chance we need that.

> Great job! Deleting the omnibus lm3-classifier file might have been jumping the gun a bit (would do that last so you can keep referencing it) Looks like creating new classifier files from here out will be much easier!

**Comment by ankurshukla03** [comment]
@diegocasmo you are right get_train_test_split is used in every notebook with the classifier. It's better to make a separate file like preprocess.py so we don't have to make a same cell in every notebook. Returning this ``(train_data, train_labels, test_data, test_labels)`` will be much better. I will try to do all that in a python file and use that in a notebook.

> As a matter of fact, it might be even cooler if instead of returning (train_df, test_df) from get_train_test_split, return instead (train_data, train_labels, test_data, test_labels). This way we don't have to unpack those values every time (this logic also seems to be repeated across all notebooks)

**Comment by tristaaan** [comment]
What is `preprocess_2` and does it replace `preprocess`?

**Comment by ankurshukla03** [comment]
`preprocess_2` is for lm2 dataset and `preprocess` is for lm3 dataset. Both are for prerpocessing but for different dataset.
> What is preprocess_2 and does it replace preprocess?

**Comment by tristaaan** [comment]
aha, they should be renamed to represent such. e.g. :  `preprocess_lm3`
`preprocess_lm2`

If the methods are similar enough you could make it single file and just
check the dimensionality of the input and append each time too but maybe
that's pedantic.

-Tristan Wright

**Comment by ankurshukla03** [comment]
- Done
> aha, they should be renamed to represent such. e.g. :  `preprocess_lm3`
`preprocess_lm2`

**Review by diegocasmo — CHANGES_REQUESTED** [review]
- [x] Where's the data being normalized here? (or did we decided to leave it out?)
- [x] Why are the ``.csv`` files placed in the same dir as the code? I thought the whole idea behind the ``/data`` dir was to have all data related files in there

**Comment by ankurshukla03** [comment]
@diegocasmo 
 - `.csv` files are still in `/data` folder. I have changed the path just to run the notebook in my laptop that's why i wrote comment before reading file in each notebook.
- all normalization part for `lm2` and `lm3` file are being done in `preprocess_lm3` and `preprocess_lm2` file
